contracts/MarketOracle.sol

```diff
@@ -56,12 +65,51 @@ contract MedianOracle is Ownable, IOracle {
         minimumProviders = minimumProviders_;
     }
 
+//    function pushReport(uint256 payload) external
+//    {
+//        address providerAddress = msg.sender;
+//        Report[2] memory reports = providerReports[providerAddress];
+//
+//        require(reports[0].timestamp > 0);
+//
+//        uint8 index_recent = reports[0].timestamp >= reports[1].timestamp ? 0 : 1;
+//        uint8 index_past = 1 - index_recent;
+//
+//        uint256 nowTimestamp = now;
+//
+//        // Check the push is not too soon after the last one.
+//        require(reports[index_recent].timestamp.add(reportSecurityDelaySec) <= nowTimestamp);
+//
+//        providerReports[providerAddress][index_past].timestamp = nowTimestamp;
+//        providerReports[providerAddress][index_past].payload = payload;
+//    }
+
     function pushReport(uint256 payload) external
     {
-        address sender = msg.sender;
-        require(providerReports[sender].timestamp > 0);
-        providerReports[sender].timestamp = now;
-        providerReports[sender].payload = payload;
+        address providerAddress = msg.sender;
+        Report[2] storage reports = providerReports[providerAddress];
+        uint256[2] memory timestamps = [reports[0].timestamp, reports[1].timestamp];
+
+        require(timestamps[0] > 0);
+
+        uint8 index_recent = timestamps[0] >= timestamps[1] ? 0 : 1;
+        uint8 index_past = 1 - index_recent;
+
+        uint256 nowTimestamp = now;
```

Comment:
Is copying this more efficient than just using `now`?

Otherwise, I think it's clearer to just use that.